test/sharness/t0180-corenet.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+test_description="Test experimental corenet commands"
+
+. lib/test-lib.sh
+
+# start iptb + wait for peering
+test_expect_success 'init iptb' '
+  iptb init -n 2 --bootstrap=none --port=0
+'
+
+startup_cluster 2
+
+test_expect_success 'peer ids' '
+  PEERID_0=$(iptb get id 0) &&
+  PEERID_1=$(iptb get id 1)
+'
+
+test_expect_success 'generate test data' '
+  dd if=/dev/urandom of=corenet0.bin bs=1K count=1 &&
+  dd if=/dev/urandom of=corenet1.bin bs=1K count=1
+'
+
+# netcat (nc) is needed for the following tests
+test_expect_success "nc is available" '
+  type nc >/dev/null
+'
+
+test_expect_success 'start ipfs listener' '
+  ipfsi 0 exp corenet listen corenet-test /ip4/127.0.0.1/tcp/10001 2>&1 > listener-stdouterr.log
+'
+
+test_expect_success 'Test server to client communications' '
+  dd if=corenet0.bin | nc -l -p 10001 &
+
+  ipfsi 1 exp corenet dial $PEERID_0 corenet-test /ip4/127.0.0.1/tcp/10002 2>&1 > dialer-stdouterr.log
+
+  nc -v 127.0.0.1 10002 | dd of=client.out
+'
+
+test_expect_success 'Test server to client communications' '
+  nc -l -p 10001 | dd of=server.out &
```

Comment:
Always collect PID of background jobs to: 1. check their return code, 2. try killing them when tests are done.